.github/workflows/publish-docker.yml

```diff
@@ -0,0 +1,77 @@
+name: Publish Docker
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: tag (ex. v0.8.3) to publish on docker
+        required: true
+
+jobs:
+  tag-docker:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        image: ["tanssi", "container-chain-simple-template", "container-chain-evm-template"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3 
+        with:
+          fetch-depth: 0
+      - name: Login to DockerHub
+        uses: docker/login-action@v2.2.0  
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - run: |
```

Comment:
it would be better to split these into smaller sub steps so that its easier to trouble shoot, understand and maintain. e.g. `checkout`, `login to dockerhub` `download files` `build and push` etc